angularjs-portal-home/src/main/webapp/partials/marketplace-details.html

```diff
@@ -1,3 +1,106 @@
+			    <span class="fa {{::portlet.faIcon}}" ng-show="portlet.faIcon != null"></span>
+			    <span class="portlet-title">{{::portlet.title}} Details</span>
+			  </h2>
+			  <a href="#/apps" class="back-to-home"><i class="fa fa-arrow-circle-o-left"></i> Back to Apps</a>
+	  	</div>
+			  
+			<div class="col-xs-12 col-lg-6">
+			 	<div class="action-buttons">
+
+			    <!-- Add to Home button with access -->
+			    <button class="btn btn-primary fname-{{::portlet.fname}}"
+			            ng-click="marketplaceCtrl.addToHome($index , portlet)"
+			            ng-if="portlet.canAdd && !portlet.hasInLayout"
+			            ng-class="{disabled : portlet.hasInLayout}">
+			      <span class="fa fa-plus"></span> Add to Favorites
+			    </button>
+
+			    <!-- Already in Home -->
+			    <button class="btn btn-disabled fname-{{::portlet.fname}}"
+			            ng-if="portlet.canAdd && portlet.hasInLayout"
+			            ng-class="{'btn-disabled' : portlet.hasInLayout}"
+			            popover="This app is already in your favorites"
+			            popover-trigger="mouseenter"
+			            popover-placement="top"
+			            popover-popup-delay="500">
+			      <i class="fa fa-check"></i> Already in Favorites
+			    </button>
+
+			    <!-- Add to Home button without access -->
+			    <button class="btn btn-disabled" ng-hide="portlet.canAdd" popover="You do not have access to this app" popover-trigger="mouseenter" popover-placement="top" popover-popup-delay="500">
+			      <span class="fa fa-plus"></span> Add to Favorites
+			    </button>
+
+			    <!-- Launch button with access -->
+			    <a class="btn btn-success" href="{{::portlet.maxUrl}}" target="{{::portlet.target}}" ng-if="portlet.canAdd"><span class="fa fa-arrow-circle-o-right"></span> Launch</a>
+
+			    <!-- Launch button without access -->
+			    <a class="btn btn-disabled" href ng-hide="portlet.canAdd" popover="You do not have access to this app" popover-trigger="mouseenter" popover-placement="top" popover-popup-delay="500"><span class="fa fa-arrow-circle-o-right"></span>Launch</a>
+			   </div>
+			</div>
+		</div>
+		<div class="row no-margin">
+			<div class="col-xs-12 col-sm-4 col-lg-3 desc-item">
+				<h3 class="center">Description</h3>
+				<p class="app-description">{{::portlet.description}}</p>
+			</div>
+			<div class="col-xs-12 col-sm-4 col-lg-3 desc-item center rating-div">
+				<h3>Ratings</h3>
+				<span><rating ng-model="portlet.rating" readonly="true" class="rating"></rating>
+	    ( {{::portlet.userRated}} rating<span ng-if="portlet.userRated !== 1">s</span> )
+	  		</span><br>
+	  		<button ng-click="openRating('sm',portlet.fname, portlet.title)" class="btn btn-default rate-button">Rate</button>
+			</div>
+			<div class="col-xs-12 col-sm-4 col-lg-3 desc-item center">
+				<h3>Related Apps</h3>
+	      <ul>
+	        <li ng-repeat="related in portlet.relatedPortlets"><a href="#/apps/search/{{ ::related.title }}">{{ ::related.title }}</a></li>
```

Comment:
Once the app-specific URL works, should the related app titles link to the entry for that related app rather than to search for the app by name?  (I think very much yes -- enabling the related apps link to behave in this way was a big motivator for this story for me. :smile: )  But I readily defer to Design on this detail, especially if I get a good shot at articulating why I think this should work the way I think it should. :smile: .